### ISSUE TITLE
Some characters are inserted left of the character to the left of the cursor

### ISSUE THREAD

**Issue by clara-avnet** [microsoft/vscode#93232]
When you type a "special" character, it is not inserted at the position of the cursor but at one position to the left. That means that it appears between the second-to-last and the last character typed before the special character.

Affected characters (probably non-exhaustive list): @, {, }, [, ], |, ~, \

- VSCode Version: 1.43.0
- OS Version: Linux x64 4.19.0-8-amd6

Steps to Reproduce:

1. Open any file with VS Code
2. Type some alphanumeric characters, e.g. "some characters"
3. At the end of the line, type some special characters, e.g. "[]~|"
4. Each of the "special" characters is inserted to the left of the character behind which the cursor is at the moment of typing. The resulting line with the example strings used above is "some character|~][s".

Does this issue occur when all extensions are disabled?: Yes

**Comment by clara-avnet** [comment]
> 
> 
> (Experimental duplicate detection)
> Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
> 
>     * [Editor Automatically Replacing Special Characters. (#53495)](https://www.github.com/microsoft/vscode/issues/53495)

No, it is not covered by that issue or any other one that I could find

**Comment by gjsjohnmurray** [comment]
What Language Mode are your files using?

**Comment by clara-avnet** [comment]
I tried it with files with different file extensions, the labels at the bottom were:
- .txt: UTF-8, LF, Nur-Text (text only)
- .md: UTF-8, LF, Markdown
- .c: UTF-8, LF, C, Linux
- .h: UTF-8, LF, C++, Linux
- .py: UTF-8, LF, Python
The problem occured in all of these files.

**Comment by alexdima** [comment]
@clara-avnet I believe this is a duplicate of #77817 or #79511

A fix is available in the Insiders Channel. Can you please try with VS Code insiders?

**Comment by clara-avnet** [comment]
I tested it with VS Code insiders (the version provided by the debian package code-insiders from 23.03.2020) before posting this issue, the problem was still there.

It is not a duplicate of #77817, because in my case the cursor is drawn at the correct position when I click at the end of the file. And if I type a "normal" character it is always typed where the cursor is. To me, it seems more like the cursor jumps left when certain characters are typed.
For #79511, it's the same. There, the cursor is always one step ahead of the text, in my case it just jumps with certain characters.

This gif illustrates the problem:
![Peek 2020-03-30 11-08](https://user-images.githubusercontent.com/56296992/77896105-08f2e700-7278-11ea-9100-3a61c133ff12.gif)

Keep in mind that I only once explicitly moved the cursor when I jumped to the end of the line. Otherwise all cursor changes were caused by typing characters.

This is with the following code insiders:
Version: 1.44.0-insider
Commit: 13b38d2a494612ace32c4fb878824d6329d4c51e
Date: 2020-03-30T08:08:20.753Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 4.19.0-8-amd64

**Comment by alexdima** [comment]
@clara-avnet I'm sorry, but I'm having trouble understanding what is presented in the GIF, because so many things are happening. Can you please use the screen casting mode (F1 > Toggle Screencast Mode) which renders the pressed keys and capture those as well in the GIF? Can you please also focus on doing 1-2 gestures, like opening a file with some "hello world" in it and then doing like 1 typing that shows the problem, so I can try to understand what is going on?

**Comment by clara-avnet** [comment]
Thanks for your patience, I hope this example is clearer:

![code-cursor-jumps2](https://user-images.githubusercontent.com/56296992/77899136-91738680-727c-11ea-9786-fc51f691047c.gif)

I don't know if it helps, but for getting a '{' on a german keyboard layout, you have to press 'Alt Gr' + '7'.

**Comment by alexdima** [comment]
@clara-avnet Are you sitting at a physical machine with Linux or are you using some sort of remote/virtualization technology?

**Comment by alexdima** [comment]
I'm asking because this feels similar to https://github.com/microsoft/vscode/wiki/Keybinding-Issues#troubleshoot-linux-keybindings

![image](https://user-images.githubusercontent.com/5047891/77899782-59207800-727d-11ea-940b-27a1b5276dec.png)

**Comment by clara-avnet** [comment]
Yes, I am using Remote Desktop to access that Linux machine. I tried the workaround suggested in the wiki article you cited and that solves the problem both for the standard VS Code installation and the insiders installation.
Thank you for your time!
